frontend/src/components/ProductDetail.jsx

```diff
@@ -15,86 +20,203 @@ function ProductDetail({ product, onBack }) {
     )
   }
 
-  const handleAddToCart = () => {
-    // TODO: Implement add to cart functionality
-    setAddedToCart(true)
-    setTimeout(() => setAddedToCart(false), 2000)
+  function makeProductNameNice(name) {
+    const words = name.split(' ')
+    const capitalizedWords = []
+
+    for (let i = 0; i < words.length; i++) {
+      const word = words[i]
+      const capitalizedWord = word.charAt(0).toUpperCase() + word.slice(1)
+      capitalizedWords.push(capitalizedWord)
+    }
+
+    return capitalizedWords.join(' ')
+  }
+
+  function makeBrandNameNice(brand) {
+    return brand.charAt(0).toUpperCase() + brand.slice(1)
+  }
+
+  function handleAddToFavorites() {
+    setAddedToFavorites(true)
+
+    setTimeout(() => {
+      setAddedToFavorites(false)
+    }, 2000)
+  }
+
+  function getKeyIngredients() {
+    const firstFive = []
+    for (let i = 0; i < 5 && i < product.ingredient_list.length; i++) {
```

Comment:
I mentioned previously that you shouldn't use "i" & "j". You can use something like "ingredientIndex" instead